src/stm32f0xx/gpio_it.c

```diff
@@ -0,0 +1,81 @@
+#include "gpio_it.h"
+
+#include <stdint.h>
+#include <stdlib.h>
+
+#include "exti.h"
+#include "gpio.h"
+#include "misc.h"
+#include "nvic.h"
+#include "status.h"
+#include "stm32f0xx.h"
+#include "stm32f0xx_rcc.h"
+#include "stm32f0xx_syscfg.h"
+
+#define GPIO_NUM_PINS 16
```

Comment:
If we have a device specific header then no and this is probably the solution. I use it to make checking the input args clearer. It is the same names/values as those in the implementation gpio.c since they are the same value. I could rename it to GPIO_IT_NUM_PINS or just drop it if that is preferred but the value is checking a GPIOAddress so despite the normal naming convention I went with GPIO_NUM_PINS as it made more sense to me.